src/main/java/io/github/thebusybiscuit/slimefun4/api/events/PreCanUnlockResearchEvent.java

```diff
@@ -0,0 +1,86 @@
+package io.github.thebusybiscuit.slimefun4.api.events;
+
+import io.github.thebusybiscuit.slimefun4.api.player.PlayerProfile;
+import io.github.thebusybiscuit.slimefun4.core.researching.Research;
+import io.github.thebusybiscuit.slimefun4.implementation.guide.BookSlimefunGuide;
+import io.github.thebusybiscuit.slimefun4.implementation.guide.ChestSlimefunGuide;
+import me.mrCookieSlime.Slimefun.Objects.Category;
+import me.mrCookieSlime.Slimefun.Objects.SlimefunItem.SlimefunItem;
+import org.apache.commons.lang.Validate;
+import org.bukkit.entity.Player;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+import javax.annotation.Nonnull;
+import javax.annotation.ParametersAreNonnullByDefault;
+
+/**
+ * This {@link Event} is called whenever a {@link Player} click to unlock a research.
+ * This is called called before {@link Research#canUnlock(Player)}
```

Comment:
```suggestion
 * This {@link Event} is called whenever a {@link Player} clicks to unlock a research.
 * This is called before {@link Research#canUnlock(Player)}.
```

Reply:
Will fix the typo 
As for the same code; yes I will add a method in research like Research#clickInteraction(SlimefunGuideImplementation,Player,Item) 
(Looking for a better name)

Reply:
Fixed typo and created : guideClickInteraction method in research